src/rgw/rgw_sal.h

```diff
@@ -224,7 +224,8 @@ class ObjectProcessor : public DataProcessor {
                        const char *if_match, const char *if_nomatch,
                        const std::string *user_data,
                        rgw_zone_set *zones_trace, bool *canceled,
-                       const req_context& rctx) = 0;
+                       const req_context& rctx,
+                       bool log_op = true) = 0;
```

Comment:
please don't use default arguments in virtual functions. i know there are other examples in rgw_sal.h, but our coding style guide argues against this in https://google.github.io/styleguide/cppguide.html#Default_Arguments:
> The defaults for arguments in a virtual function call are determined by the static type of the target object, and there's no guarantee that all overrides of a given function declare the same defaults.

Reply:
done